models/yield_yak/avalanche_c/yield_yak_avalanche_c_trades.sql

```diff
@@ -0,0 +1,110 @@
+{{ config(
+    alias = 'trades',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address'],
+    post_hook='{{ expose_spells(\'["avalanche_c"]\',
+                                "project",
+                                "yield_yak",
+                                \'["Henrystats"]\') }}'
+    )
+}}
+
+
+{% set project_start_date = '2021-09-15' %}
+
+WITH 
+
+dexs as (
+        SELECT 
+            evt_block_time as block_time, 
+            -- '' as taker, commenting this as there's no trader in the event 
+            '' as maker, 
+            _amountIn as token_sold_amount_raw, 
+            _amountOut as token_bought_amount_raw, 
+            CAST(NULL as double) as amount_usd, 
+            _tokenIn as token_sold_address, 
+            _tokenOut as token_bought_address, 
+            contract_address as project_contract_address, 
+            evt_tx_hash as tx_hash, 
+            '' as trace_address, 
+            evt_index
+        FROM 
+        {{ source('yield_yak_avalanche_c', 'YakRouter_evt_YakSwap') }}
+        {% if not is_incremental() %}
```

Comment:
minor point, but for your reference, `{% if not is_incremental() %}` approach isn't necessarily needed on decoded tables, as the data would only exist from project start anyway. it's most important on joins to base tables.

Reply:
will update, thank you ser!

Reply:
for transparency, sometimes i include that in my models simply so when building locally & testing, i can apply a small date timeframe to filter the decoded tables and make it all run fast -- that's preference though!